xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -0,0 +1,246 @@
+            )
+        case .other(let msg):
+            return String(
+                localized: "An unknown error occurred: \(msg)",
+                comment: "Unknown UserProfileService error description"
+            )
+        }
+    }
+}
+
+struct UserProfileContentPayload: Equatable, Hashable {
+    var profile: UserProfile
+    var statistics: UserProfileStatistics
+    var following: [StoryUser]
+    var entries: [EntryStub]
+    var isFollowingUser: Bool
+}
+
+actor UserProfileService {
+    private var noosphere: NoosphereService
+    private var database: DatabaseService
+    private var addressBook: AddressBookService
+    
+    private let logger = Logger(
+        subsystem: Config.default.rdns,
+        category: "UserProfileService"
+    )
+    
+    init(
+        noosphere: NoosphereService,
+        database: DatabaseService,
+        addressBook: AddressBookService
+    ) {
+        self.noosphere = noosphere
+        self.database = database
+        self.addressBook = addressBook
+    }
+    
+    /// Retrieve all the content for the App User's profile view, fetching their profile, notes and address book.
+    func getOwnProfile() async throws -> UserProfileContentPayload {
+        guard let nickname = AppDefaults.standard.nickname,
+              let petname = Petname(nickname) else {
+                  throw UserProfileServiceError.missingPreferredPetname
+        }
+        
+        let did = try await self.noosphere.identity()
+        let following = try await self.getFollowingList(
+            sphere: self.noosphere,
+            localAddressBook: AddressBook(sphere: self.noosphere),
+            traversalPath: .none
+        )
+        let notes = try await self.noosphere.list()
+        
+        var entries: [EntryStub] = []
+        
+        for slug in notes {
+            let slashlink = Slashlink(slug: slug)
+            let memo = try await noosphere.read(slashlink: slashlink)
+            
+            guard let memo = memo.toMemo() else {
+                continue
+            }
+
+            entries.append(
+                EntryStub(
+                    address: Slashlink(slug: slug).toPublicMemoAddress(),
+                    excerpt: memo.excerpt(),
+                    modified: memo.modified
+                )
+            )
+        }
+        
+        guard let did = Did(did) else {
+            throw UserProfileServiceError.invalidSphereIdentity
+        }
+        
+        let profile = UserProfile(
+            did: did,
+            petname: petname,
+            // TODO: replace with _profile_.json data
+            pfp: "sub_logo",
+            bio: "Wow, it's your own profile! With its own codepath!",
+            category: .you
+        )
+        
+        return UserProfileContentPayload(
+            profile: profile,
+            statistics: UserProfileStatistics(
+                noteCount: entries.count,
+                backlinkCount: -1, // TODO: populate with real count
```

Comment:
I don't know how to populate this right now so leaving as -1 since it's offensively wrong looking in the UI.